core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -565,6 +564,22 @@ private void runInternal() {
             throw e;
           } finally {
             jumpListener.setListener(listener);
+            // An extremely short deadline may expire before the stream listener is set. This causes
```

Comment:
Note: this can be _anywhere_ in this method, although after the `try {` makes it easier to reason about.

It's a bit weird to be in the `finally`, since it isn't essential it is run. Could we put it after the finally block or within the try (e.g., after `startCall()`)?